zio-http/src/main/scala/zhttp/service/server/content/handlers/ServerResponseHandler.scala

```diff
@@ -19,20 +19,24 @@ private[zhttp] trait ServerResponseHandler[R] {
 
   type Ctx = ChannelHandlerContext
 
-  def writeResponse(msg: Response, jReq: FullHttpRequest)(implicit ctx: Ctx): Unit = {
+  def writeResponse(msg: Response, jReq: HttpRequest)(implicit ctx: Ctx): Unit = {
 
     ctx.write(encodeResponse(msg))
     msg.data match {
-      case HttpData.BinaryStream(stream)  =>
-        rt.unsafeRun(ctx) {
-          writeStreamContent(stream).ensuring(UIO(releaseRequest(jReq)))
-        }
-      case HttpData.RandomAccessFile(raf) =>
-        unsafeWriteFileContent(raf())
-        releaseRequest(jReq)
-      case _                              =>
-        ctx.flush()
+      case HttpData.Asynchronous(_) =>
         releaseRequest(jReq)
+        throw new IllegalStateException("Cannot write data to response")
```

Comment:
Need to think about this. I feel we can use this to perform a high performance echo operation.